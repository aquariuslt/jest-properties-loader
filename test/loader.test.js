const _ = require('lodash');
const path = require('path');
const fs = require('fs');
const jestPropertyLoader = require('../lib');

const sampleSource = fs.readFileSync(path.join(__dirname, './__fixtures__/sample.properties')).toString();

describe('jest-property-loader', () => {

  it('# should load properties file in jest correctly', () => {
    const transformedResult = jestPropertyLoader.process(sampleSource);

    expect(transformedResult).not.toBeUndefined();
    expect(_.get(transformedResult, 'foo.bar')).toEqual('foobar');
  });
});
