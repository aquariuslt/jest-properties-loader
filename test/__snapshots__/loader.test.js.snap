// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-property-loader # should load properties file in jest correctly 1`] = `"module.exports={\\"foo\\":{\\"bar\\":\\"foobar\\"}}"`;
